server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -162,12 +165,17 @@ public void createFields(Consumer<IndexableField> onField,
                                      boolean indexed,
                                      boolean docValued,
                                      boolean stored) {
-                if (indexed) {
-                    onField.accept(new DoublePoint(name, value.doubleValue()));
+                if (indexed && docValued) {
```

Comment:
Note that these changes had already been made to the Float type; it doesn't really make any difference in functionality at the end, but it makes things a lot easier to test if the DataType and translog source parser produce the same lucene Fields.